src/LanguageServer/Protocol/Features/Options/SolutionCrawlerOptionsService.cs

```diff
@@ -10,18 +10,10 @@
 namespace Microsoft.CodeAnalysis.SolutionCrawler;
 
 [ExportWorkspaceService(typeof(ISolutionCrawlerOptionsService)), Shared]
-internal sealed class SolutionCrawlerOptionsService : ISolutionCrawlerOptionsService
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class SolutionCrawlerOptionsService(IGlobalOptionService globalOptions) : ISolutionCrawlerOptionsService
 {
-    private readonly IGlobalOptionService _globalOptions;
-
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public SolutionCrawlerOptionsService(IGlobalOptionService globalOptions)
-    {
-        _globalOptions = globalOptions;
-    }
-
     public bool EnableDiagnosticsInSourceGeneratedFiles
-        => _globalOptions.GetOption(SolutionCrawlerOptionsStorage.EnableDiagnosticsInSourceGeneratedFiles) ??
-        _globalOptions.GetOption(SolutionCrawlerOptionsStorage.EnableDiagnosticsInSourceGeneratedFilesFeatureFlag);
+        => globalOptions.GetOption(SolutionCrawlerOptionsStorage.EnableDiagnosticsInSourceGeneratedFiles);
```

Comment:
So I think we might need a bigger approach for ripcord settings like this, since I think this PR _as written_ puts is in a worse place. temporarily. Right now it's obvious this was a feature flag rollout, but once we delete it then this setting lives around even longer...and it's not obvious anymore (unless you have the context for it) that the only reason the setting exists was for a ripcord -- no user should need to change it otherwise.

I'd say we should be deleting these switches (and the commit async and document outline flags) _entirely_, or add some sort of TODO on them, or check in a docs file listing the flags and the expected removal date. Because otherwise this will still live around until somebody notices them "eventually".

Reply:
i'm ok pulling the flags entirely.